tests/integration/test_all_sets_image_diffs.py

```diff
@@ -1,159 +1,54 @@
+    str
+        The path to the results directory.
+    """
+    # Set -d flag to use the .cfg file for running additional diagnostic sets.
+    sys.argv.extend(["-d", CFG_PATH])
 
-def _compare_images(
-    mismatched_images: List[str],
-    image_name: str,
-    path_to_actual_png: str,
-    path_to_expected_png: str,
-) -> List[str]:
-    # https://stackoverflow.com/questions/35176639/compare-images-python-pil
-
-    actual_png = Image.open(path_to_actual_png).convert("RGB")
-    expected_png = Image.open(path_to_expected_png).convert("RGB")
-    diff = ImageChops.difference(actual_png, expected_png)
+    params = _get_test_params()
+    results = runner.run_diags(params)
 
-    diff_dir = f"{TEST_ROOT_PATH}image_check_failures"
-    if not os.path.isdir(diff_dir):
-        os.mkdir(diff_dir)
+    results_dir = results[0].results_dir
 
-    bbox = diff.getbbox()
-    # If `diff.getbbox()` is None, then the images are in theory equal
-    if bbox is None:
-        pass
-    else:
-        # Sometimes, a few pixels will differ, but the two images appear identical.
-        # https://codereview.stackexchange.com/questions/55902/fastest-way-to-count-non-zero-pixels-using-python-and-pillow
-        nonzero_pixels = (
-            diff.crop(bbox)
-            .point(lambda x: 255 if x else 0)
-            .convert("L")
-            .point(bool)
-            .getdata()
-        )
-        num_nonzero_pixels = sum(nonzero_pixels)
-        logger.info("\npath_to_actual_png={}".format(path_to_actual_png))
-        logger.info("path_to_expected_png={}".format(path_to_expected_png))
-        logger.info("diff has {} nonzero pixels.".format(num_nonzero_pixels))
-        width, height = expected_png.size
-        num_pixels = width * height
-        logger.info("total number of pixels={}".format(num_pixels))
-        fraction = num_nonzero_pixels / num_pixels
-        logger.info("num_nonzero_pixels/num_pixels fraction={}".format(fraction))
-
-        # Fraction of mismatched pixels should be less than 0.02%
-        if fraction >= 0.0002:
-            mismatched_images.append(image_name)
-
-            simple_image_name = image_name.split("/")[-1].split(".")[0]
-            shutil.copy(
-                path_to_actual_png,
-                os.path.join(diff_dir, "{}_actual.png".format(simple_image_name)),
-            )
-            shutil.copy(
-                path_to_expected_png,
-                os.path.join(diff_dir, "{}_expected.png".format(simple_image_name)),
-            )
-            # https://stackoverflow.com/questions/41405632/draw-a-rectangle-and-a-text-in-it-using-pil
-            draw = ImageDraw.Draw(diff)
-            (left, upper, right, lower) = diff.getbbox()
-            draw.rectangle(((left, upper), (right, lower)), outline="red")
-            diff.save(
-                os.path.join(diff_dir, "{}_diff.png".format(simple_image_name)),
-                "PNG",
-            )
+    logger.info(f"results_dir={results_dir}")
```

Comment:
Instead of running e3sm_diags using subprocess, we call `run_diags()` directly. This means real-time stderr outputs, ability to interactively walk through the code for debugging, and faster failures (if they happen). 

Before we had to wait for their subprocess to finish which was frustrating whenever there are failures because it would take a long time.